tests/resources/test_branch.py

```diff
@@ -406,3 +407,35 @@ def test_experiment_data_source_no_project_id(session):
         branch.experiment_datasource
 
     assert not session.calls
+
+
+def test_get_by_root_id(session, collection, branch_path):
+    # Given
+    branches_data = BranchDataFactory.create_batch(1)
+    session.set_response({'response': branches_data})
+    root_id = uuid.uuid4()
+
+    # When
+    branch = collection.get_by_root_id(branch_root_id=root_id)
+
+    # Then
+    assert session.calls == [FakeCall(
+        method='GET',
+        path=branch_path,
+        params={'page': 1, 'per_page': 1, 'root': str(root_id), 'version': 'latest'}
```

Comment:
Should this just be testing for `root` and `version` rather than the full call structure?  I feel like those are the only bits that actually matter, and `page` and `per_page` are internal implementation details.

Reply:
Only `root` and `version` are relevant, yes. But the `FakeCall` construct compares the whole payload, not just present keys, so they must be included. And since this is a unit test, including internal details isn't a problem (although it should of course be avoided when possible).